portfolio/src/main/webapp/script.js

```diff
@@ -16,13 +16,83 @@
  * Adds a random greeting to the page.
  */
 function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+  const facts =
+      ["I'm left-handed.", "My right eye has 20/20 vision, while my left eye has 20/100 vision.", "My favorite flavor of ice cream is mint chocolate chip.", "I like to listen to the Beatles.", "My favorite color is purple.", "I want to visit Antarctica someday.", "I like to watch let's play videos of horror games."];
 
   // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  const factContainer = document.getElementById('random-fact-container');
+  factContainer.innerText = fact;
 }
+
+/* Shows the previous image in the gallery. */
+function previousImg() {
+    const images = ["images/ico1.jpg", "images/ico2.jpg", "images/fountain.jpg", "images/palm-tree.jpg", "images/ice-cream.jpeg", "images/sunset.jpg"];
+
+    idx = parseInt(document.querySelector('#galleryimg').dataset.idx);
+    prev_idx = (idx - 1 + images.length) % images.length;
+    document.querySelector('#galleryimg').src = images[prev_idx];
+    document.querySelector('#galleryimg').dataset.idx = prev_idx;
+}
+
+/* Shows the next image in the gallery. */
+function nextImg() {
+    const images = ["images/ico1.jpg", "images/ico2.jpg", "images/fountain.jpg", "images/palm-tree.jpg", "images/ice-cream.jpeg", "images/sunset.jpg"];
+
+    idx = parseInt(document.querySelector('#galleryimg').dataset.idx);
+    console.log(idx + 1)
+    next_idx = (idx + 1) % images.length;
+    document.querySelector('#galleryimg').src = images[next_idx];
+    document.querySelector('#galleryimg').dataset.idx = next_idx;
+}
+
+/* Shows or hides the Mandelbrot fractal. */
+function selectMandelbrot() {
```

Comment:
One thing I noticed is that selectMandelbrot, selectSierpinski, and selectKoch all have pretty much the same exact functionality except the only thing that changes is the source image and the alt text. Do you think there's a way you can abstract out this common functionality so that it can be more reusable? For example, say if you wanted to add another GIF, currently you'd have to create an entirely new function with the same functionality. Try to think about this a bit more and if you come up with a better solution, feel free to send out another PR, but in the meantime this is fine. Good job!